spec/support/shared_examples/requests/post_form.rb

```diff
@@ -20,6 +21,21 @@
         post post_request_path, empty_form_request_body
         expect(response.code).to eq("200")
       end
+
+      it "includes validation errors for the form data" do
+        invalid_form_data.each do |invalid_data|
+          post post_request_path, form_factory => invalid_data.merge(token: correct_form.token)
+
+          invalid_form = build(form_factory, **invalid_data)
+          invalid_form.validate
+
+          invalid_form.errors.messages.values.flatten.each do |error_message|
+            # We include error messages twice, but rspec have no built-in an include twice method yet.
+            # Hence, the scan will make sure we matche the message twice in the rendered page.
```

Comment:
```suggestion
            # Hence, the scan will make sure we match the message twice in the rendered page.
```

Also, in this case, why not `= 2`?

Reply:
It is because in some views the same error message text appears as page text for explaining to the user what to do. Hence, in order to keep the maintainability of this test low, I have decided that testing that we show "at least 2" of those error messages is good enough :)